plugins/cri/images/plugin.go

```diff
@@ -96,39 +96,32 @@ func init() {
 			} else {
 				return nil, fmt.Errorf("failed to find snapshotter %q", defaultSnapshotter)
 			}
-			var snapshotRoot string
-			if plugin := ic.Plugins().Get(plugins.SnapshotPlugin, defaultSnapshotter); plugin != nil {
-				snapshotRoot = plugin.Meta.Exports["root"]
-			}
-			if snapshotRoot == "" {
-				// Try a root in the same parent as this plugin
-				snapshotRoot = filepath.Join(filepath.Dir(ic.Properties[plugins.PropertyRootDir]), plugins.SnapshotPlugin.String()+"."+defaultSnapshotter)
+
+			snapshotRoot := func(snapshotter string) (snapshotRoot string) {
+				if plugin := ic.Plugins().Get(plugins.SnapshotPlugin, snapshotter); plugin != nil {
+					snapshotRoot = plugin.Meta.Exports["root"]
+				}
+				if snapshotRoot == "" {
+					// Try a root in the same parent as this plugin
+					snapshotRoot = filepath.Join(filepath.Dir(ic.Properties[plugins.PropertyRootDir]), plugins.SnapshotPlugin.String()+"."+snapshotter)
+				}
+				return snapshotRoot
 			}
-			options.ImageFSPaths[defaultSnapshotter] = snapshotRoot
-			log.L.Infof("Get image filesystem path %q for snapshotter %q", snapshotRoot, defaultSnapshotter)
+
+			options.ImageFSPaths[defaultSnapshotter] = snapshotRoot(defaultSnapshotter)
+			log.L.Infof("Get image filesystem path %q for snapshotter %q", options.ImageFSPaths[defaultSnapshotter], defaultSnapshotter)
 
 			for runtimeName, rp := range config.RuntimePlatforms {
 				snapshotter := rp.Snapshotter
 				if snapshotter == "" {
 					snapshotter = defaultSnapshotter
-				} else if _, ok := options.ImageFSPaths[snapshotter]; !ok {
-					if s, ok := options.Snapshotters[defaultSnapshotter]; ok {
-						options.Snapshotters[defaultSnapshotter] = s
-					} else {
-						return nil, fmt.Errorf("failed to find snapshotter %q", defaultSnapshotter)
-					}
```

Comment:
I am not sure if this is only needed for `defaultSnapshotter` (if so this is noop and we can remove since we've done this at line 94-98.

Or what we really want is ?

```golang
					if s, ok := allSnapshotters[snapshotter]; ok {
						options.Snapshotters[snapshotter] = s
					} else {
						return nil, fmt.Errorf("failed to find snapshotter %q", snapshotter)
					}
```